tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java

```diff
@@ -1519,6 +1531,175 @@ public boolean onOptionsItemSelected(MenuItem item) {
+
+                                               } catch (EMP_Exception e) {
+                                                   return e;
+                                               }
+                                               return null;
+                                           }
+                                           @Override
+                                           protected void onPostExecute(final EMP_Exception e) {
+                                               if (e == null) {
+                                                   MainActivity.this.oMenu.findItem(R.id.action_mirrorcache_subscribe).setEnabled(true);
+
+                                               } else if (e != null) {
+                                                   Log.e(TAG, e.getMessage(), e);
+                                                   new AlertDialog.Builder(MainActivity.this).setTitle("ERROR:").setMessage(e.getMessage()).create().show();
+                                                   MainActivity.this.oMenu.findItem(R.id.action_mirrorcache_unsubscribe).setEnabled(true);
+                                               }
+                                           }
+                                       }.execute();
+                                   }
+                               }).create().show();
+                return true;
+            }
+            case R.id.action_mirrorcache_addproduct: {
+                MainActivity.this.oMenu.findItem(R.id.action_mirrorcache_addproduct).setEnabled(false);
+
+                try {
+                    mc.addProduct(MainActivity.this.oRootOverlay);
+                    MainActivity.this.oMenu.findItem(R.id.action_mirrorcache_deleteproduct).setEnabled(true);
+
+                } catch (EMP_Exception e) {
+                    Log.e(TAG, e.getMessage(), e);
+                    new AlertDialog.Builder(MainActivity.this).setTitle("ERROR:").setMessage(e.getMessage()).create().show();
+                    MainActivity.this.oMenu.findItem(R.id.action_mirrorcache_addproduct).setEnabled(true);
+                }
+                return true;
+            }
+            case R.id.action_mirrorcache_deleteproduct: {
+                MainActivity.this.oMenu.findItem(R.id.action_mirrorcache_deleteproduct).setEnabled(false);
+
+                try {
+                    mc.removeProduct(MainActivity.this.oRootOverlay, false);
+                    MainActivity.this.oMenu.findItem(R.id.action_mirrorcache_addproduct).setEnabled(true);
+
+                } catch (EMP_Exception e) {
+                    Log.e(TAG, e.getMessage(), e);
+                    new AlertDialog.Builder(MainActivity.this).setTitle("ERROR:").setMessage(e.getMessage()).create().show();
+                    MainActivity.this.oMenu.findItem(R.id.action_mirrorcache_deleteproduct).setEnabled(false);
+                }
+                return true;
+            }
```

Comment:
There are instructions in test-vandv. Take a look at AddRemoveGetTest.java class level comments.